src/main/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestworker/services/GeneratePdfService.kt

```diff
@@ -159,12 +159,19 @@ class GeneratePdfService {
   }
 
   fun preProcessData(input: Any?): Any? {
+    if (input is List<*> && input.isEmpty()) {
+      return "No information has been recorded"
+    }
     if (input is Map<*, *>) {
       // If it's a map, process the key
       val returnMap = mutableMapOf<String, Any?>()
       val inputKeys = input.keys
       inputKeys.forEach { key ->
-        returnMap[processKey(key.toString())] = preProcessData(input[key]) as Any?
+        if (input[key] == null || input[key] == "null") {
```

Comment:
This part can also be removed if the other comment's refactor is done, since we just call preProcessData with the value and we don't need to worry about any special cases here.